src/nextjournal/clerk/viewer.cljc

```diff
@@ -180,7 +182,119 @@
       (with-viewer :html
         (into (mkup-fn node) (cond text [text] content (map with-md-viewer content)))))))
 
-(declare !viewers)
+;; EDN writer
+(defn var->data [v] (wrapped-with-viewer v))
+
+#_(var->data #'var->data)
+
+(defn fn->str [f]
+  (let [pr-rep (pr-str f)
+        f-name (subs pr-rep (count "#function[") (- (count pr-rep) 1))]
+    f-name))
+
+#_(fn->str (fn []))
+#_(fn->str +)
+
+;; TODO: consider removing this and rely only on viewers
+(defn make-readable [x]
```

Comment:
Maybe we can just drop this altogether, as now describe should make both functions and vars readable by processing such types with their relevant viewers. But I'd propose we change this in a follow up step.

We might still need the meta removal step.